base/intfuncs.jl

```diff
@@ -387,6 +354,10 @@ julia> nextpow(4, 16)
 See also [`prevpow`](@ref).
 """
 function nextpow(a::Real, x::Real)
+    # Special case fast path for x::Integer, a == 2.
+    # This is a very common case. Constant prop will make sure that a call site
+    # specified as `nextpow(2, x)` will get this special case inlined.
+    a == 2 && isa(x, Integer) && return _nextpow2(x)
     a <= 1 && throw(DomainError(a, "`a` must be greater than 1."))
     x <= 0 && throw(DomainError(x, "`x` must be positive."))
```

Comment:
I think this error check needs to come before the check for `a == 2`. That's causing the test failures, since `_nextpow2(x)` won't throw a `DomainError` for `x < 1`.

Reply:
Which does present a problem though, because the old `nextpow2` and `prevpow2` functions did not throw an error for `x == 0` and returned the next negative power of two for negative integers.

Reply:
That seems like a marginal and debatable behavior that we could break. Or if it’s really useful, we could generalize it. But I suspect it’s not. Negative powers of two are in fact not powers of two at all.

Reply:
I'll just move the <= 0 check to the top for now; seems like the conservative thing to do.